rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend

```diff
@@ -0,0 +1,173 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.rcp4.views.tcltestrun
+
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+import java.io.OutputStream
+import java.nio.charset.Charset
+import java.nio.file.Files
+import java.util.HashMap
+import java.util.Map
+import javax.inject.Inject
+import org.eclipse.core.resources.IProject
+import org.eclipse.core.runtime.IProgressMonitor
+import org.eclipse.jdt.junit.JUnitCore
+import org.eclipse.jface.viewers.IStructuredSelection
+import org.eclipse.jface.viewers.LabelProvider
+import org.eclipse.jface.window.Window
+import org.eclipse.ui.PlatformUI
+import org.eclipse.ui.dialogs.ElementListSelectionDialog
+import org.slf4j.LoggerFactory
+import org.testeditor.dsl.common.ui.utils.ProgressMonitorRunner
+import org.testeditor.rcp4.tcltestrun.TclGradleLauncher
+import org.testeditor.rcp4.tcltestrun.TclLauncher
+import org.testeditor.rcp4.tcltestrun.TclMavenLauncher
+import org.testeditor.tcl.dsl.ui.testlaunch.Launcher
+
+class TclLauncherUi implements Launcher {
+	static val logger = LoggerFactory.getLogger(TclLauncherUi)
+
+	@Inject ProgressMonitorRunner progressRunner
+	@Inject TclMavenLauncher mavenLauncher
+	@Inject TclGradleLauncher gradleLauncher
+
+	override boolean launch(IStructuredSelection selection, IProject project, String elementId, String mode,
+		boolean parameterize) {
+		val options = new HashMap<String, Object>
+		if (project.getFile("build.gradle").exists) {
+			return launchTest(selection, project, elementId, gradleLauncher, options)
+		}
+		if (project.getFile("pom.xml").exists) {
+			if (parameterize) {
+				val mavenOptions = uiCollectMavenOptions(project)
+				if (mavenOptions == null) {
+					return true // should an option always be selected?
+				}
+				options.putAll(mavenOptions)
+			}
+			return launchTest(selection, project, elementId, mavenLauncher, options)
+		}
+		logger.warn("gradle based launching test for tcl element='{}' failed, since file='build.gradle' was not found.",
+			elementId)
+		logger.warn("maven based launching test for tcl element='{}' failed, since file='pom.xml' was not found.",
+			elementId)
+		return false
+	}
+
+	private def Map<String, Object> uiCollectMavenOptions(IProject project) {
```

Comment:
Why do we return a map here when it's really just a value that is returned by the dialog?